codegen/templates/http_client.tmpl

```diff
@@ -170,35 +182,43 @@ func (c *{{$clientName}}) {{title .Name}}(
 	switch res.StatusCode {
 		case {{.OKStatusCode.Code}}:
 			var responseBody {{unref .ResponseType}}
+			{{if isPointerType .ResponseType -}}
 			err = res.ReadAndUnmarshalBody(&responseBody)
+			{{else -}}
+			rawBody, err := res.ReadAll()
+			if err != nil {
+				return _res, respHeaders, err
+			}
+			err = json.Unmarshal(rawBody, &responseBody)
```

Comment:
I dont like this naked `json.Unmarshal` without having error logging. Can we add a new method to `res` that's like `ReadAndUnmarshalNonPointerValue()` or something.